### PR TITLE
feat: add a continuous integration pipeline

### LINKED ISSUE #35: Implement a continuous integration pipeline
- This is outlined in the work from #8
 - Should be the last thing done after all other set-up has been completed

### PR DESCRIPTION
**Pull request by JQrdan**
Includes:
 - Run build
 - Run unit and e2e tests
 - Deploy storybook to github pages on merge to master
 - Create bundle size report (as a comment on PR)
 - Create test coverage report (as a comment on PR)
 - Upload storybook artifact on PR
 - Lint commit messages
 - Lint all code
 - Mark stale issues and remove dead ones

Closes: #35 

Signed-off-by: Jordan <jordan.tucker1@ibm.com>

### CODE REVIEW

File: .github/workflows/git-commit-lint.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    name: Lint Commit Messages
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Use Node.js 12.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
```
Comment (nictownsend) [inline .github/workflows/git-commit-lint.yml]: 14 now :)

File: .github/workflows/node-deploy-storybook.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      - master
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    name: Deploy Storybook
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 12.x
```
Comment (nictownsend) [inline .github/workflows/node-deploy-storybook.yml]: 14 :)

File: .github/workflows/node-pr-jobs.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        with:
+          ref: master
+      - name: Install Dependencies
+        run: npm ci
+      - name: Build
+        run: npm run build
+      - name: Upload Master Bundle Report
+        run: echo "MASTER_REPORT=$(cat generated/bundle-analyser/report.json)" >> $GITHUB_ENV
+      - uses: actions/checkout@v2
+      - name: Use Node.js 12.x
```
Comment (nictownsend) [inline .github/workflows/node-pr-jobs.yml]: 14 :)

**Review by nictownsend — CHANGES_REQUESTED** [review]
Other than node versions LGTM

File: .github/actions/commenter/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Copyright Strimzi authors.
+# License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+FROM node:slim
```
Comment (matthew-chirgwin) [inline .github/actions/commenter/Dockerfile:3]: Any reason why there is a dockerfile per action? Or could there be a simlink? Just thinking of saving upkeep

Comment (matthew-chirgwin) [reply .github/actions/commenter/Dockerfile:3]: Or have the actions refer to one dockerfile build?

**Review by matthew-chirgwin — CHANGES_REQUESTED** [review]
Looks good - just one question re dockerfiles :)

**Review by dlabaj — APPROVED** [review]

**Review by scholzj — APPROVED** [review]

File: .github/actions/commenter/Dockerfile (hunk 4 above)
Comment (JQrdan) [reply .github/actions/commenter/Dockerfile:3]: I haven't played around with it but actions are supposed to be their own little packages. I don't think we want to but these could each live in their own repo and we'd just change the location of where the action is pulled from

**Review by matthew-chirgwin — APPROVED** [review]

**Pull request merged**